### ISSUE TITLE
Add hint_range to vec2 in shaders.

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot-proposals#4790]
### Describe the project you are working on

Various vfx - Godot's particle system

### Describe the problem or limitation you are having in your project

Currently in 4.x, we have two floats per min/max range of random properties of the particles.
While they work as expected by dragging the sliders around, i think a custom widget for the editor should exist for when we want to set min and max ranges of value.
This is for both shaders, but also for exported variables.
Right now, the particles material implements a completely custom solution to have the sliders move together. It should probably be standardised and made reusable across the editor.
Without implementing a custom solution every time, it's not possible to have nice to use range sliders that support a min and a max.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

See above

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add a hint_double_range annotation to Vector2 and vec2 in shaders.
`uniform vec2 my_range : hint_range(min, max, step) = vec2(defaut_min, default_max)` 

Add a widget that looks like the following.

![doublerange](https://user-images.githubusercontent.com/7917475/176859892-76af7921-a992-4364-bab2-684aacd5c24e.png)

It's essentially two widgets for range one on top of the other, with a toggle button to move them together or not.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes, a few lines of script per each script where you want to implement a range, plus an editor plugin.
It's cumbersome and there's probably enough places where we would benefit from it.

### Is there a reason why this should be core and not an add-on in the asset library?

This is an engine modification.

**Comment by RPicster** [comment]
I like the idea as I don't think the two sliders right now have a nice ux.
Maybe this could also be an alternative for the element:
![image](https://user-images.githubusercontent.com/9423774/176867174-f14aaf3d-2f67-4dc8-9e3e-596e1f0e12d9.png)

**Comment by winston-yallow** [comment]
I'm not sure about the name `hint_range` as it is already used for specifying in which range a float or int can be provided, not for actually getting a range from the user. Maybe something like `hint_rangevector` to differentiate them better? Other than that I really would like to have this hint

**Comment by lostminds** [comment]
> I'm not sure about the name `hint_range` as it is already used for specifying in which range a float or int can be provided

I agree with this, and in GDScript we have `@export_range` that also specifies a single value within a range. Maybe `hint_range_span`, as it's still a slider range with a min/max, but a span with two values in the range. 

If something like this is implemented as a shader hint I think it would also be a good idea to add the same `@export_xx` functionality in GDScript.
